presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -381,11 +387,44 @@ private PrestoSparkTaskSourceRdd createTaskSourcesRdd(
 
     private static SetMultimap<Integer, ScheduledSplit> assignSourceDistributionSplits(Session session, List<ScheduledSplit> splits)
     {
-        int taskCount = getSparkInitialPartitionCount(session);
-        checkArgument(taskCount > 0, "taskCount must be greater then zero: %s", taskCount);
+        boolean splitSizeMissing = splits.stream()
+                .anyMatch(split -> !split.getSplit().getConnectorSplit().getSplitSizeInBytes().isPresent());
         ImmutableSetMultimap.Builder<Integer, ScheduledSplit> result = ImmutableSetMultimap.builder();
-        for (int splitIndex = 0; splitIndex < splits.size(); splitIndex++) {
-            result.put(splitIndex % taskCount, splits.get(splitIndex));
+        if (splitSizeMissing) {
+            int taskCount = getSparkInitialPartitionCount(session);
+            checkArgument(taskCount > 0, "taskCount must be greater then zero: %s", taskCount);
+            for (int splitIndex = 0; splitIndex < splits.size(); splitIndex++) {
+                result.put(splitIndex % taskCount, splits.get(splitIndex));
+            }
+        }
+        else {
+            int minSplitsPerSparkPartition = getMinSplitsPerSparkPartition(session);
```

Comment:
add `checkArgument(minSplitsPerSparkPartition > 0 , "minSplitsPerSparkPartition must be greater than zero: %s", minSplitsPerSparkPartition)`